src/lls_model.jl

```diff
@@ -72,6 +72,31 @@ function jac_residual(nls :: LLSModel, x :: AbstractVector)
   end
 end
 
+function jac_structure_residual(nls :: LLSModel)
+  if isa(nls.A, AbstractLinearOperator)
+    error("Jacobian is a LinearOperator. Use `jac_op_residual` instead.")
+  elseif nls.A isa AbstractSparseMatrix
+    return findnz(nls.A)[1:2]
+  else
+    m, n = size(nls.A)
+    I = ((i,j) for i = 1:m, j = 1:n)
+    return (getindex.(I, 1)[:], getindex.(I, 2)[:])
+  end
+end
+
+function jac_coord_residual!(nls :: LLSModel, x :: AbstractVector, rows :: AbstractVector{Int}, cols :: AbstractVector{Int}, vals :: AbstractVector)
+  increment!(nls, :neval_jac_residual)
+  if isa(nls.A, AbstractLinearOperator)
+    error("Jacobian is a LinearOperator. Use `jac_op_residual` instead.")
```

Comment:
Is this necessary? There will be an error on line 94.

Reply:
I tested what would be error on line 94, and was surprised to find this huge log:
```
A = sprand(10, 10, 0.5)
v = zeros(3)
op = LinearOperator(A)
v[2] = op[2,3]
```
```
ERROR: MethodError: Cannot `convert` an object of type LinearOperator{Float64,getfield(LinearOperators, Symbol("##36#39"))
{LinearOperator{Float64,getfield(LinearOperators, Symbol("##36#39"))
{LinearOperator{Int64,getfield(LinearOperators, Symbol("##155#157"))
{Array{Int64,1}},getfield(LinearOperators, Symbol("##156#158"))
{Int64,Array{Int64,1}},getfield(LinearOperators, Symbol("##156#158"))
{Int64,Array{Int64,1}}},LinearOperator{Float64,getfield(LinearOperators, Symbol("##10#13")){SparseMatrixCSC{Float64,Int64}},getfield(LinearOperators, 
Symbol("##11#14")){SparseMatrixCSC{Float64,Int64}},getfield(LinearOperators, 
Symbol("##12#15")){SparseMatrixCSC{Float64,Int64}}}},getfield(LinearOperators, 
Symbol("##37#40")){LinearOperator{Float64,getfield(LinearOperators, 
Symbol("##10#13")){SparseMatrixCSC{Float64,Int64}},getfield(LinearOperators, 
Symbol("##11#14")){SparseMatrixCSC{Float64,Int64}},getfield(LinearOperators, 
Symbol("##12#15...
[many lines later]
...1}}}}}}} to an object of type Float64
```
Not sure what happened here. But on the issue at hand, the error message in L90 is nicer.

Reply:
Oh boy. An explicit Julia error message. Thanks.